### ISSUE TITLE
Is it just SMTP with some new headers?

### ISSUE THREAD

**Issue by snipe** [snipe/ssmtp#2]
I kinda don't think so. SMTP by nature is permissive and open, and we're trying to turn that on its ear, raising the barrier for shenanigans, spoofing, and spam. We can add headers to SMTP all day, but if the behavior of SMTP isn't changed, the outcome doesn't change.

**Comment by mgdm** [comment]
I'd agree with that, if for no other reason than it'd be great to prevent the massive waste of bandwidth and processing that having to deal with spam represents. I don't have any bright ideas at the moment for how to go about something better, but I definitely think we shouldn't be looking to base on SMTP in its current state.

**Comment by snipe** [comment]
That's sort of what I'm thinking. "If we could start over, what would our business requirements be, given what we know about the state of email, spamming and spoofing today, and where trends indicate it's likely to go?"

**Comment by Andorbal** [comment]
I agree that extending SMTP will probably not get much traction.  Things like DKIM and SPF should help validate a sender, but because it's not implemented by everyone, you can't rely on it.  A completely new system would mean that any implementations of it would, by definition, include all the features that make it better.  That said, it would still need to inter-operate with SMTP or you couldn't get much adoption because there wasn't already a huge user base.

Maybe a server that spoke the new language could try to send a message using it first, then fall back on SMTP if it wasn't available?

**Comment by snipe** [comment]
I think that's the direction we were thinking. Ultimately, we want it to be backwards compatible to some degree or we'll have half the internet unable to talk to the other half, which is dumb. But I feel like there has to be a good, "urgent" business reason to force slow-moving enterprise systems to get on board. Yes, they'll save money because they won't have to manage the kinds of elaborate (or third-party, expensive) spam prevention systems, but we all know how slowly that machine moves.

**Comment by zanyterp** [comment]
Probably not new headers but an extension (a la HTTP 1.0 vs 1.1)? Or start with user-level implementation, either as a stand-alone client or plug-in, and then move to server- or (new)protocol-based  implementation? This will allow users to be early adopters as-desired but still leave room for growth. For me, I would prefer something I can setup for myself as a user before rolled out enterprise-level.
